packages/http/src/route.ts

```diff
@@ -216,28 +216,43 @@ export function setRoute(context: DecoratorContext, entity: Type, details: Route
   const state = context.program.stateMap(routesKey);
 
   if (state.has(entity) && entity.kind === "Namespace") {
-    const existingValue: RoutePath = state.get(entity);
-    if (existingValue.path !== details.path) {
+    const existingPath: string | undefined = state.get(entity);
+    if (existingPath !== details.path) {
       reportDiagnostic(context.program, {
         code: "duplicate-route-decorator",
         messageId: "namespace",
         target: entity,
       });
     }
   } else {
-    state.set(entity, details);
+    state.set(entity, details.path);
+    if (entity.kind === "Operation" && details.shared) {
+      setSharedRoute(context.program, entity as Operation);
+    }
   }
 }
 
+const sharedRoutesKey = createStateSymbol("sharedRoutes");
+
+export function setSharedRoute(program: Program, operation: Operation) {
+  program.stateMap(sharedRoutesKey).set(operation, true);
+}
+
 export function isSharedRoute(program: Program, operation: Operation): boolean {
-  return program.stateMap(routesKey).get(operation)?.shared;
+  return program.stateMap(sharedRoutesKey).get(operation) === true;
 }
 
 export function getRoutePath(
   program: Program,
   entity: Namespace | Interface | Operation
 ): RoutePath | undefined {
-  return program.stateMap(routesKey).get(entity);
+  const path = program.stateMap(routesKey).get(entity);
+  return path
+    ? {
+        path,
+        shared: entity.kind === "Operation" && isSharedRoute(program, entity as Operation),
```

Comment:
The original logic simply plumbed details through without regard to whether it was an operation, interface, etc. so this change would introduce a behavioral change.

Reply:
In the original design, applying `@route` on a namespace or interface does not propagate the sharedness status down to the contained operations, so it never worked in practice.  The new decorator just makes this explicit in that `@sharedRoute` can't be applied anywhere but to an operation.